File: src/main/java/io/koju/autopos/service/ItemService.java
package io.koju.autopos.service;

import io.koju.autopos.domain.Item;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Service Interface for managing Item.
 */
public interface ItemService {

    /**
     * Save a item.
     * @return the persisted entity
     */
    public Item save(Item item);

    /**
     *  get all the items.
     *  @return the list of entities
     */
    public Page<Item> findAll(Pageable pageable);

    /**
     *  get the "id" item.
     *  @return the entity
     */
    public Item findOne(Long id);

    /**
     *  delete the "id" item.
     */
    public void delete(Long id);
}

File: scalang/src/main/scala/com/kapilkoju/autopos/catalog/service/ItemService.scala
package com.kapilkoju.autopos.catalog.service

import java.lang.Long
import javax.annotation.{Nonnull, Nullable}

import com.kapilkoju.autopos.catalog.domain.Item
import org.springframework.data.domain.{Page, Pageable}

trait ItemService {

  def save(item: Item): Item

  @Nonnull
  def findAll(queryTerm: String, pageable: Pageable): Page[Item]

  @Nullable
  def getItem(id: Long): Item

  @Nullable
  def getItemWithDetail(id: Long): Item

  def delete(id: Long)

  def adjustQuantity(item: Item, number: Integer)

  def addQuantity(item: Item, number: Integer) {
    adjustQuantity(item, number)
  }

  def subtractQuantity(item: Item, number: Integer) {
    adjustQuantity(item, -number)
  }
}
